### PR TITLE
new files

### PR DESCRIPTION
**Pull request by emrecaydin**
thanks by advance

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -29,5 +29,5 @@ APP_SECRET=09ee2f37b2c71db0759693180af6f077
 # For an SQLite database, use: "sqlite:///%kernel.project_dir%/var/data.db"
 # For a PostgreSQL database, use: "postgresql://db_user:db_password@127.0.0.1:5432/db_name?serverVersion=11&charset=utf8"
 # IMPORTANT: You MUST configure your server version, either here or in config/packages/doctrine.yaml
-DATABASE_URL=mysql://db_user:db_password@127.0.0.1:3306/db_name
+DATABASE_URL=mysql://emre2:3673@127.0.0.1:3306/menu
```
Comment (axelhamil) [inline .env]: You need to create an .env.local file and put your login details there, you don't have to put them in the .env

File: assets/js/app.js [hunk 2]
```diff
@@ -9,4 +9,13 @@
...
+import $ from 'jquery';
+var section = $('li');
+
+function toggleAccordion() {
+    section.removeClass('active');
+    $(this).addClass('active');
+}
+
+section.on('click', toggleAccordion);
+
```
Comment (axelhamil) [inline assets/js/app.js]: In this file, you just have to uncomment "import $ from 'jquery;'.
You don't have to write your program inside

File: assets/js/carte.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import $ from 'jquery';
```
Comment (axelhamil) [inline assets/js/carte.js]: This line is only necessary in assets/js/app.js

File: src/Controller/StaticPagesController.php [hunk 4]
```diff
@@ -38,4 +40,15 @@ public function indexCgu(): Response
...
+    /**
+     * @Route("/carte", name="carte")
+     * @return Response
+     */
+    public function carteMenu(): Response
+    {
+        $repository = $this->getDoctrine()->getRepository(Category::class);
+        $categories = $repository->findWithProducts();
+        return $this->render('static/carte.html.twig', ['categories' => $categories]);
+    }
```
Comment (axelhamil) [inline src/Controller/StaticPagesController.php]: carteMenu is not a staticPages create a controller for him.

**Review by axelhamil — CHANGES_REQUESTED** [review]
your css must be in public / style /
and not in assets / style / app.scss, we agreed with the other members for that.
Also resolves conflicts.

File: .env [hunk 5]
```diff
@@ -26,7 +26,7 @@ APP_SECRET=09ee2f37b2c71db0759693180af6f077
 
 ###> doctrine/doctrine-bundle ###
 # Format described at https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/configuration.html#connecting-using-a-url
-# For an SQLite database, use: "sqlite:///%kernel.project_dir%/var/data.db"
+# For an SQLit###< doctrine/doctrine-bundle ###e database, use: "sqlite:///%kernel.project_dir%/var/data.db"
 # For a PostgreSQL database, use: "postgresql://db_user:db_password@127.0.0.1:5432/db_name?serverVersion=11&charset=utf8"
 # IMPORTANT: You MUST configure your server version, either here or in config/packages/doctrine.yaml
 DATABASE_URL=mysql://emre2:3673@127.0.0.1:3306/menu
```
Comment (axelhamil) [inline .env]: You must not put your login details in the .env file but in the .env.local

File: assets/js/app.js [hunk 6]
```diff
@@ -9,13 +9,4 @@
...
-var section = $('li');
-
-function toggleAccordion() {
-    section.removeClass('active');
-    $(this).addClass('active');
-}
-
-section.on('click', toggleAccordion);
-
+// import $ from 'jquery';
```
Comment (axelhamil) [inline assets/js/app.js]: if you use Jquery you must uncomment this line

File: assets/js/carte.js [hunk 7]
```diff
@@ -1,9 +1,10 @@
 import $ from 'jquery';
```
Comment (axelhamil) [inline assets/js/carte.js]: You don't need this line if you uncommented it in the app.js file

File: src/Controller/CarteController.php [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class CarteController extends AbstractController
+{
+    /**
+     * @Route("/carte", name="carte")
+     */
+    public function index(): Response
```
Comment (axelhamil) [inline src/Controller/CarteController.php]: the name of this function is not adequate, instead use a name such as "menuMap", "carteMenu" ...Etc

File: src/Controller/CarteController.php [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        /** @var CategoryRepository $repository */
+        $repository = $this->getDoctrine()->getRepository(Category::class);
+        $categories = $repository->findWithProducts();
```
Comment (axelhamil) [inline src/Controller/CarteController.php:20]: why did you use this function? "Finby" would have been enough? no?

File: templates/carte/index.html.twig [hunk 10]
```diff
@@ -0,0 +1,67 @@
+{% extends "static/layout.html.twig" %}
```
Comment (axelhamil) [inline templates/carte/index.html.twig]: index.html.twig is not a properly chosen name.
This template must not extend static / layout, you must create a /layout.html.twig which extends /base.html.twig in your map / folder

File: src/Repository/CategoryRepository.php [hunk 11]
```diff
@@ -19,6 +19,9 @@ public function __construct(ManagerRegistry $registry)
         parent::__construct($registry, Category::class);
     }
 
+    /**
+     * @return int|mixed|string
+     */
     public function findWithProducts()
```
Comment (axelhamil) [inline src/Repository/CategoryRepository.php:25]: Why are you using this function? use findby. No?

**Review by axelhamil — CHANGES_REQUESTED** [review]
Check comment, some files are wrongly named and files are misplaced

File: src/Controller/CarteController.php (hunk 9 above)
Comment (emrecaydin) [reply src/Controller/CarteController.php:20]: no because i'm using left join in this function

File: src/Repository/CategoryRepository.php (hunk 11 above)
Comment (emrecaydin) [reply src/Repository/CategoryRepository.php:25]: no

File: public/style/carte.css [hunk 12]
```diff
@@ -14,6 +14,7 @@ body {
     text-rendering: optimizelegibility;
     margin: 0;
     padding: 0;
+    background-color:black!important;
```
Comment (axelhamil) [inline public/style/carte.css]: what is "!important" ?

**Review by axelhamil — CHANGES_REQUESTED** [review]

File: public/style/carte.css (hunk 12 above)
Comment (emrecaydin) [reply public/style/carte.css]: by default the background is transparent, so my page's background color is white also the links are too. To make it seems better i used "!important"

**Review by axelhamil — DISMISSED** [review]

**Review by Mgaren — DISMISSED** [review]

**Review by axelhamil — APPROVED** [review]

**Review by Mgaren — APPROVED** [review]

File: src/Controller/CarteController.php [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+class CarteController extends AbstractController
+{
+    /**
+     * @Route("/carte", name="carte")
+     */
+    public function carteMenu(): Response
+    {
+        /** @var CategoryRepository $repository */
+        $repository = $this->getDoctrine()->getRepository(Category::class);
```
Comment (RomainClair) [inline src/Controller/CarteController.php:19]: You should use dependency injection to access your services in controllers

**Review by RomainClair — APPROVED** [review]

**Pull request merged**